extension/storage/dbstorage/extension.go

```diff
@@ -35,6 +35,19 @@ func newDBStorage(logger *zap.Logger, config *Config) (extension.Extension, erro
 
 // Start opens a connection to the database
 func (ds *databaseStorage) Start(context.Context, component.Host) error {
+	if ds.driverName == driverSQLiteLegacy {
+		// Log warning about legacy driver usage
+		ds.logger.Warn("Legacy driver 'sqlite3' is used, please review documentation to update your configuration")
+		// Change legacy driver to a new one
+		ds.driverName = driverSQLite
+		// Try to convert legacy driver options and log errors if any
+		var err error
+		ds.datasourceName, err = replaceCompatDSNOptions(ds.datasourceName)
+		if err != nil {
```

Comment:
should this error be returned?

Reply:
There are 2 types of errors produced by `replaceCompatDSNOptions` - informational and critical (that probably should stop component loading). I'll try to refactor this method to properly segregate them

Reply:
Updated, now returned error is a hard-stop error and returned to caller